lib/storage/index_db.go

```diff
@@ -1699,6 +1707,14 @@ func (th *topHeap) Pop() any {
 	panic(fmt.Errorf("BUG: Pop shouldn't be called"))
 }
 
+// TopHeapMetricNameEntry represents an entry from `top heap` used in stats for series metric names.
+type TopHeapMetricNameEntry struct {
```

Comment:
not implying anything, but would it be easier to keep it as is, but add a new field to `TSDBStatus` instead:
```
type TSDBStatus struct {
...
MetricNameStatss [string]MetricNameStats
}
```
That should have all the keys from `SeriesCountByMetricName ` fields. And it also could be nil, if that functionality isn't enabled. So client could easily determine whether it has access to this feature and don't display it if no data available.

Reply:
It's possible to add a new field MetricNameStatRecords []Record. But it'll add extra complications:
1) during tsdb status merge it must be kept in sync with `SeriesCountByMetricName` by topN entries
2) it must have the same order as SeriesCountByMetricName for access by idx at frontend
3) it complicates frontend table merge logic.